File: src/main/java/dev/efnilite/ipp/mode/single/TimeTrialMode.java
package dev.efnilite.ipp.mode.single;

import dev.efnilite.ip.leaderboard.Leaderboard;
import dev.efnilite.ip.menu.community.SingleLeaderboardMenu;
import dev.efnilite.ipp.generator.single.TimeTrialGenerator;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public final class TimeTrialMode extends SingleMode {

    private final Leaderboard leaderboard = new Leaderboard(getName(), Leaderboard.Sort.TIME);

    @Override
    public void create(Player player) {
        create(player, TimeTrialGenerator::new);
    }

    @Override
    public Leaderboard getLeaderboard() {
        return leaderboard;
    }

    @Override
    public @NotNull String getName() {
        return "time_trial";
    }
}

File: src/main/kotlin/dev/efnilite/iep/mode/TimeTrialMode.kt
package dev.efnilite.iep.mode

import dev.efnilite.iep.config.Config
import dev.efnilite.iep.generator.TimeTrialGenerator
import dev.efnilite.iep.leaderboard.Leaderboard

object TimeTrialMode : Mode {

    override val name = "time trial"

    override val leaderboard = Leaderboard(
        name,
        Config.CONFIG.getDouble("mode-settings.time-trial.score"),
        Leaderboard.Sort.TIME
    )

    override fun getGenerator() = TimeTrialGenerator()
}